### PR TITLE
feat(swap): add SwapDepositAddressReady event handler [WEB-288]

### PR DESCRIPTION
**Pull request by GabrielBuragev**

### CODE REVIEW

File: packages/swap/src/event-handlers/swapDepositAddressReady.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+const swapDepositAddressReadyArgs = z.object({
+  depositAddress: encodedAddress,
+  destinationAddress: encodedAddress,
+  sourceAsset: chainflipAssetEnum,
+  destinationAsset: chainflipAssetEnum,
+  channelId: u64,
+});
+
+export type SwapDepositAddressReadyArgs = z.input<
```
Comment (jerryafr) [inline packages/swap/src/event-handlers/swapDepositAddressReady.ts:14]:
```suggestion
export type SwapDepositAddressReadyEvent = z.input<
```

File: packages/swap/src/event-handlers/__tests__/swapDepositAddressReady.test.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    );
+
+    expect(swapDepositChannel).toMatchObject({
+      channelId: expect.any(BigInt),
+      issuedBlock: expect.any(Number),
+      srcChain: expect.any(String),
+    });
+  });
+
+  describe('if deposit channel already exists', () => {
```
Comment (zoheb391) [inline packages/swap/src/event-handlers/__tests__/swapDepositAddressReady.test.ts:38]: probably a redundant describe

**Review by zoheb391 — APPROVED** [review]
nice work!

File: packages/swap/src/event-handlers/swapDepositAddressReady.ts (hunk 1 above)
Comment (GabrielBuragev) [reply packages/swap/src/event-handlers/swapDepositAddressReady.ts:14]: Good point!

File: packages/swap/src/event-handlers/swapDepositAddressReady.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  const {
+    depositAddress,
+    destinationAddress,
+    sourceAsset,
+    destinationAsset,
+    channelId,
+    ...rest
+  } = swapDepositAddressReadyArgs.parse(event.args);
+
+  await prisma.swapDepositChannel.upsert({
```
Comment (acdibble) [inline packages/swap/src/event-handlers/swapDepositAddressReady.ts:34]: We should probably change the other insertion into an upsert too

Comment (GabrielBuragev) [reply packages/swap/src/event-handlers/swapDepositAddressReady.ts:34]: Which one? Can you point me to it please?

Comment (GabrielBuragev) [reply packages/swap/src/event-handlers/swapDepositAddressReady.ts:34]: Which one? Can you point me to it please?

Comment (GabrielBuragev) [reply packages/swap/src/event-handlers/swapDepositAddressReady.ts:34]: Done

**Review by acdibble — APPROVED** [review]

**Pull request merged**